### PR TITLE
Issue #3

### PR DESCRIPTION
**Pull request by ventos**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -17,6 +17,7 @@ var (
 )
 
 func main() {
+
```
Comment (rossmeier) [inline main.go]: ?

File: database/repository.go [hunk 2]
```diff
@@ -1,7 +1,103 @@
...
+		if err != nil {
+			return nil, errors.New("Error while reading tar (DbScratch)")
+		}
+		if pkg.FileInfo().IsDir() {
+			continue
+		}
+		io.Copy(buf, reader)
+		str, err := buf.ReadString('\n')
+		if err != nil {
+			panic(err)
```
Comment (rossmeier) [inline database/repository.go]: return wrapped error instead of panicing

File: database/repository.go [hunk 3]
```diff
@@ -1,7 +1,103 @@
...
+		}
+		if str != "%FILENAME%\n" {
+			panic("Invalid filename designator: " + str)
```
Comment (rossmeier) [inline database/repository.go]: ditto

File: database/repository.go [hunk 4]
```diff
@@ -1,7 +1,103 @@
...
+		}
+		filename, err := buf.ReadString('\n')
+		if err != nil {
+			panic(err)
```
Comment (rossmeier) [inline database/repository.go]: ditto

File: database/repository.go [hunk 5]
```diff
@@ -1,7 +1,103 @@
...
+		}
+		p, err := packet.FromFilename(filename)
+		if err != nil {
+			panic(err)
```
Comment (rossmeier) [inline database/repository.go]: ditto

File: database/repository.go [hunk 6]
```diff
@@ -1,7 +1,103 @@
...
+		}
+
+		readDb = append(readDb, *p)
```
Comment (rossmeier) [inline database/repository.go]: call a callback here

**Review by rossmeier — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (rossmeier) [reply main.go]: still, wtf?

File: database/repository.go [hunk 7]
```diff
@@ -1,7 +1,91 @@
...
+
+	ParseDBgunzipped(r, func(p *packet.Packet) {
+		readDb = append(readDb, *p)
+	})
+
+	return readDb, nil
+}
+
+// ParseDBgunzipped reads a pacman .db file and creats a []DatabaseEntry
+func ParseDBgunzipped(r io.Reader, cb func(*packet.Packet)) error {
```
Comment (rossmeier) [inline database/repository.go]: GUnzipped (proper pascal case)

File: database/repository.go [hunk 8]
```diff
@@ -1,7 +1,91 @@
...
+	if err != nil {
+		return nil, errors.Wrap(err, "Error gunzipping file")
+	}
+	defer zr.Close()
+
+	return ParseDBgunzippedSlice(zr)
+}
+
+// ParseDBgunzippedSlice reads a pacman .db file and creats a []DatabaseEntry
+func ParseDBgunzippedSlice(r io.Reader) ([]packet.Packet, error) {
```
Comment (rossmeier) [inline database/repository.go]: GUnzipped

**Review by rossmeier — CHANGES_REQUESTED** [review]
Just minor things

File: main.go (hunk 1 above)
Comment (ventos) [reply main.go]: still not resolved, ha? :monkey_face:

**Review by rossmeier — CHANGES_REQUESTED** [review]
LGTM, please rebase on master for merge

**Review by rossmeier — APPROVED** [review]

**Comment by rossmeier** [comment]
Got squash-merged on the cli

**Pull request closed without merging**